src/main/java/build/buildfarm/worker/Worker.java

```diff
@@ -124,10 +126,30 @@ private static ManagedChannel createChannel(String target) {
     return builder.build();
   }
 
-  public Worker(WorkerConfig config) {
+  private static Path getValidRoot(WorkerConfig config) throws ConfigurationException {
+    String rootValue = config.getRoot();
+    // Have to handle empty string here because all later APIs happily ignore
+    // empty strings (WTF)
+    if (Strings.isNullOrEmpty(rootValue)) {
+        throw new ConfigurationException("root value in config missing");
+    }
+    return Paths.get(rootValue);
+  }
+
+  private static Path getValidCasCacheDirectory(WorkerConfig config, Path root) throws ConfigurationException {
+    String casCacheValue = config.getCasCacheDirectory();
+    // Have to handle empty string here because all later APIs happily ignore
+    // empty strings (WTF)
```

Comment:
Can we clean up this comment and the one above? Also, APIs for what? Path#resolve?

Reply:
Not just `Path` resolve. Literally everything that this data gets used on totally ignores empty strings. But you are right - since we have the test, we may also remove the comment, since the Test should catch regressions.